### PR TITLE
add image scan method

### PR DESCRIPTION
**Pull request by codeWithUtkarsh**
The Golang application scan Docker images for vulnerability and BOM, and generate reports corresponding to them.

### CODE REVIEW

**Review by markjacksonfishing — CHANGES_REQUESTED** [review]
Can you give a description that encompasses what this body of work represents

**Comment by markjacksonfishing** [comment]
Also, opening an issue that corresponds to this pr is the happy path.

**Comment by codeWithUtkarsh** [comment]
Added description that can help understand the purpose of the code. @markyjackson-taulia Does that look good?

**Comment by markjacksonfishing** [comment]
Question: Why no use a scanning GitHub action such Anchore?

**Comment by codeWithUtkarsh** [comment]
The objective was not just the vulnerability scan. There are other requirements(dependency tree, licenses, etc) too which we will be adding to the capability of this application.  Apart from that, this will be a microservice piece integrated with the Ortelius dashboard operating out of the box. I don't know more capability about Anchore Scan Action but I guess it is limited to vulnerability scan only.

I'll read more on the Anchore Scan though.

**Comment by markjacksonfishing** [comment]
With regards to the other requirements, are those being tracked anywhere?
My reasons for asking these questions is to better understand the PR and cross reference the requirements to:
- Determine if the code is extensible enough to fit the requirements
- Determine if the code is extensible enough to fit the future architecture
- And just to make sure artifacts are tracked

**Comment by sbtaylor15** [comment]
Resolves ortelius/ortelius#275

**Comment by sbtaylor15** [comment]
[Service Catalog Requirements](https://docs.google.com/document/d/1wS_aiQhmyJjUwPrQhxzEkSdmHD04kAgD-hABEmAT_aU/edit?usp=sharing)

File: src/config.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+port = "8080"
+containername= "tmp-container"
+username= "1645370"
+password= "<password>"
```
Comment (markjacksonfishing) [inline src/config.toml]: Please add a line feed

File: src/functions/docker-exec-util.go [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
```
Comment (markjacksonfishing) [inline src/functions/docker-exec-util.go:19]: This really is a dup of line 20 import. By pulling in `github.com/docker/docker/api/types/containers` you are already pulling in `github.com/docker/docker/api/types`

File: src/functions/docker-exec-util.go [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+	// Define ports to be exposed (has to be same as hostconfig.portbindings.newport)
+	exposedPorts := map[natting.Port]struct{}{
+		newport: {},
+	}
+
+	// Configuration
+	// https://godoc.org/github.com/docker/docker/api/types/container#Config
+	config := &container.Config{
+		Image:        imagename,
+		Env:          []string{"DB_HOST=192.168.225.51", "DB_PORT=9876"},
```
Comment (markjacksonfishing) [inline src/functions/docker-exec-util.go]: I would prefer we not hardcode addresses

**Review by markjacksonfishing — CHANGES_REQUESTED** [review]
I also do not see any test in any of this PR. I would prefer we at the very least test using mock

**Comment by markjacksonfishing** [comment]
I also converted this PR to draft while you work through the test

File: src/config.toml (hunk 1 above)
Comment (codeWithUtkarsh) [reply src/config.toml]: added a line feed

File: src/functions/docker-exec-util.go (hunk 3 above)
Comment (codeWithUtkarsh) [reply src/functions/docker-exec-util.go]: eliminated hardcoded values

File: src/functions/docker-exec-util.go (hunk 2 above)
Comment (codeWithUtkarsh) [reply src/functions/docker-exec-util.go:19]: @markyjackson-taulia I was reading and trying to remove these dup. But I found that this is the standard followed by Golang. The types will have their own functions and struct and types/containers will have their own functions and structs. Golang suggests importing the package whose function and structs that we want to use. This way it removes the over-fetching of packages that the user might not need. Let me know what you think.

**Review by markjacksonfishing — APPROVED** [review]
/lgtm

**Pull request merged**